### PR TITLE
Make cs2cs support 4D coordinates.

### LINKED ISSUE #1354: cs2cs has problems with time-dependent transforms
Performing time-dependant transformations with `cs2cs` does not seem to work correctly. I have observed two problems which may or or may not be connected. Both observations are related to transformations between ITRFyyyy and ETRFyyyy:

#### Observation 1: ITRF2014 (EPSG:7789) -> ETRF2014 (EPSG:8401):

`projinfo` shows to matches:

```
$ projinfo -s EPSG:7789 -t EPSG:8401 -o PROJ
Candidate operations found: 2
-------------------------------------
Operation n°1:

EPSG:8366, ITRF2014 to ETRF2014 (1), 0.0 m, Europe - ETRS89

PROJ string:
+proj=helmert +x=0 +y=0 +z=0 +rx=0 +ry=0 +rz=0 +s=0 +dx=0 +dy=0 +dz=0 +drx=8.5e-05 +dry=0.000531 +drz=-0.00077 +ds=0 +t_epoch=1989 +convention=position_vector

-------------------------------------
Operation n°2:

EPSG:8880, ITRF2014 to ETRF2014 (2), 0.0 m, Europe - ETRS89

PROJ string:
+proj=helmert +x=0 +y=0 +z=0 +rx=0.001785 +ry=0.011151 +rz=-0.01617 +s=0 +dx=0 +dy=0 +dz=0 +drx=8.5e-05 +dry=0.000531 +drz=-0.00077 +ds=0 +t_epoch=2010 +convention=position_vector
```

When using it with `cs2cs` it appears to be a null transform:
```
$ echo 3496737.2679    743254.4507  5264462.9620 2019.0 | cs2cs -f %.5f EPSG:7789 EPSG:8401
3496737.26790	743254.45070 5264462.96200 2019.0
```

However, manually specifying the two candiate operations with `cct` gives actual results:

```
# candiate 1:
$ echo 3496737.2679    743254.4507  5264462.9620 2019.0 | cct +proj=helmert +x=0 +y=0 +z=0 +rx=0 +ry=0 +rz=0 +s=0 +dx=0 +dy=0 +dz=0 +drx=8.5e-05 +dry=0.000531 +drz=-0.00077 +ds=0 +t_epoch=1989 +convention=position_vector
 3496737.7577    743253.9940  5264462.7011     2019.0000

# candidate 2:
$ echo 3496737.2679    743254.4507  5264462.9620 2019.0 | cct +proj=helmert +x=0 +y=0 +z=0 +rx=0.001785 +ry=0.011151 +rz=-0.01617 +s=0 +dx=0 +dy=0 +dz=0 +drx=8.5e-05 +dry=0.000531 +drz=-0.00077 +ds=0 +t_epoch=2010 +convention=position_vector
 3496737.7577    743253.9940  5264462.7011     2019.0000
```

So, it would seem `cs2cs` or maybe `proj_create_crs_to_crs()` is doing something wrong with the time input.

#### Observation 2: ITRF2000 (EPSG:4896) -> ETRF2005 (EPSG:7930)

Here it is clear that the time input isn't used, as demonstrated with the two calls to `cs2cs` below that return the same output on the xyz-dimensions:

```
$ echo 3496737.2679    743254.4507  5264462.9620 2019.0 | cs2cs -f %.4f EPSG:4896 EPSG:7930
3496737.4924	743254.3308 5264462.8221 2019.0

$ echo 3496737.2679    743254.4507  5264462.9620 2010.0 | cs2cs -f %.4f EPSG:4896 EPSG:7930
3496737.4924	743254.3308 5264462.8221 2010.0
```

Let's see what `projinfo` says and if `cct` agrees with `cs2cs`:

```
$ ../../../cmake/bin/projinfo -s EPSG:4896 -t EPSG:7930 -o PROJ
Candidate operations found: 1
-------------------------------------
Operation n°1:

EPSG:7950, ITRF2005 to ETRF2000 (1), 0.0 m, Europe - ETRS89

PROJ string:
+proj=helmert +x=0.0541 +y=0.0502 +z=-0.0538 +rx=0.000891 +ry=0.00539 +rz=-0.008712 +s=0.0004 +dx=-0.0002 +dy=0.0001 +dz=-0.0018 +drx=8.1e-05 +dry=0.00049 +drz=-0.000792 +ds=8e-05 +t_epoch=2000 +convention=position_vector

$ echo 3496737.2679    743254.4507  5264462.9620 2019.0 | ./cct +proj=helmert +x=0.0541 +y=0.0502 +z=-0.0538 +rx=0.000891 +ry=0.00539 +rz=-0.008712 +s=0.0004 +dx=-0.0002 +dy=0.0001 +dz=-0.0018 +drx=8.1e-05 +dry=0.00049 +drz=-0.000792 +ds=8e-05 +t_epoch=2000 +convention=position_vector
 3496737.7857    743254.0394  5264462.6437     2019.0000
# output from cs2cs was:
#3496737.4924	 743254.3308  5264462.8221     2019.0 
```

Again there's some problem with handling time correctly.

### PR DESCRIPTION
**Pull request by kbevers**
This is a bit of a hack, 4D coordinates *will* be written to STDOUT
but the output format speficied with -f is not respected for the
t component, rather it is forward verbatim from the input.

Fixes #1354

### CODE REVIEW

File: src/apps/cs2cs.cpp [hunk 1]
```diff
@@ -113,6 +114,16 @@ static void process(FILE *fid)
...
         z = strtod(s, &s);
 
+        /* To avoid breaking existing tests, we read what is a possible t    */
+        /* component of the input and rewind the s-pointer so that the final */
+        /* output has consistant behaviour, with or without t values.        */
+        /* This is a bit of a hack, 4D coordinates will be written to STDOUT */
+        /* but the output format speficied with -f is not respected for the  */
+        /* t component, rather it is forward verbatim from the input.        */
+        char *before_time = s;
+        time = strtod(s, &s);
```
Comment (rouault) [inline src/apps/cs2cs.cpp]: could be declared and initialized in the same line: double time = ...
Actually I'm wondering if time will not shadow the time() function.

**Review by rouault — COMMENTED** [review]
Perhaps a test in test/cli/testvarious

File: src/apps/cs2cs.cpp (hunk 1 above)
Comment (kbevers) [reply src/apps/cs2cs.cpp]: I've changed it, just in case

**Comment by kbevers** [comment]
> Perhaps a test in test/cli/testvarious

Done.

**Comment by kbevers** [comment]
To be honest, this is not an ideal fix but it will work for the foreseeable future. The proper fix is to rewrite `cs2cs` more or less completely and have it be 4D by default. A good candidate for a breaking change introduced with PROJ 7.

**Pull request merged**